### PR TITLE
fix(Share): #MAG-340 fix a bug where you cant lower rights

### PR DESCRIPTION
**Pull request by Rseuret**
## Describe your changes
fix a bug where you cant remove lower rights on a folder

## Issue ticket number and link

https://jira.support-ent.fr/browse/MAG-340

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/java/fr/cgi/magneto/service/impl/DefaultFolderService.java [hunk 1]
```diff
@@ -328,23 +330,23 @@ private Future<JsonObject> preDeleteFolders(List<String> folderIds) {
...
-                                 .map(String.class::cast)
-                                 .filter(bo -> !myBoards.contains(bo))
-                                 .collect(Collectors.toList());
-                         foldersUpdate.add(updateOldFolder(newBoardsIds));
-                     });
-                     FutureHelper.all(foldersUpdate)
-                             .onSuccess(s-> promise.complete(new JsonObject()))
-                             .onFailure(r->promise.fail(r.getMessage()));
-                 }).onFailure(r -> promise.fail(r.getMessage()))).onFailure(r -> promise.fail(r.getMessage()));
+                        .onSuccess(res -> {
```
Comment (Rseuret) [inline src/main/java/fr/cgi/magneto/service/impl/DefaultFolderService.java:335]: indentation

File: src/main/java/fr/cgi/magneto/service/impl/DefaultFolderService.java [hunk 2]
```diff
@@ -613,7 +615,7 @@ public Future<JsonObject> updateOldFolder(List<String> boardIds) {
     public Future<JsonObject> getFolderByBoardId(String boardId) {
         Promise<JsonObject> promise = Promise.promise();
         MongoQuery query = new MongoQuery(this.collection);
-        query.match(new JsonObject().put(Field.BOARDIDS, new JsonObject().put(Mongo.IN,new JsonArray().add(boardId))));
+        query.match(new JsonObject().put(Field.BOARDIDS, new JsonObject().put(Mongo.IN, new JsonArray().add(boardId))));
```
Comment (Rseuret) [inline src/main/java/fr/cgi/magneto/service/impl/DefaultFolderService.java:618]: indentation

File: src/main/java/fr/cgi/magneto/service/impl/DefaultFolderService.java [hunk 3]
```diff
@@ -626,9 +628,11 @@ public Future<JsonObject> getFolderByBoardId(String boardId) {
                 } else {
                     promise.complete(new JsonObject());
                 }
-            }}));
+            }
```
Comment (Rseuret) [inline src/main/java/fr/cgi/magneto/service/impl/DefaultFolderService.java:631]: indentation

File: src/main/java/fr/cgi/magneto/service/impl/DefaultFolderService.java [hunk 4]
```diff
@@ -202,6 +202,7 @@ private Future<List<String>> getBoardIdsInFolders(List<String> folderIds) {
 
     /**
      * Returns list of folder ids that are only my folders
+     *
```
Comment (Rseuret) [inline src/main/java/fr/cgi/magneto/service/impl/DefaultFolderService.java:205]: indentation

File: src/main/java/fr/cgi/magneto/service/impl/DefaultFolderService.java [hunk 5]
```diff
@@ -261,7 +262,7 @@ private Future<JsonObject> preDeleteFoldersWithChildren(List<String> folderIds,
         Promise<JsonObject> promise = Promise.promise();
 
         if (restore) {
-             getFolderChildrenIdsOwnerOnly(folderIds, ownerId)
+            getFolderChildrenIdsOwnerOnly(folderIds, ownerId)
```
Comment (Rseuret) [inline src/main/java/fr/cgi/magneto/service/impl/DefaultFolderService.java:265]: indentation

**Review by alicedraillard — APPROVED** [review]

**Pull request merged**